### ISSUE TITLE
Java Troubleshooting Tutorial: NegativeArraySizeException

### ISSUE THREAD

**Issue by ltagliaferri** [sourcegraph/learn#311]
Write a short tutorial on Java's `NegativeArraySizeException` error message, following the "[How to troubleshoot Python ZeroDivisionError](https://learn.sourcegraph.com/how-to-troubleshoot-python-zerodivisionerror)" example tutorial.

The tutorial should be organized as follows:

```
---
title: How to troubleshoot Java error: NegativeArraySizeException
author: your name
tags: [tutorial, Java, troubleshooting]
publicationDate: today's date
description: Learn how to error handle the Java error: NegativeArraySizeException
image: https://storage.googleapis.com/sourcegraph-assets/learn/headers/sourcegraph-learn-header.png
imageAlt: Sourcegraph Learn
browserTitle: NegativeArraySizeException in Java error handling
type: posts
---

Introduction with the error message

## Reproducing the error

Show how the error can be created.

## Solution 1

Provide a possible solution.

## Solution 2

Provide another solution.

## (optional) Solution 3

Optionally provide a third solution. 

## Learn more

Search across open source Java repositories that have the `NegativeArraySizeException` to understand the message more.

<SourcegraphSearch query=NegativeArraySizeException lang:java" patternType="literal"/>

Check out more Sourcegraph Learn tutorials on [Java](https://learn.sourcegraph.com/tags/java).
```

If this seems like a tutorial that you can write effectively for the developer community, please follow our [Hacktoberfest 2021 contributor guidelines](https://github.com/sourcegraph/learn/blob/main/docs/hacktoberfest-2021.md) to proceed.

**Comment by Ln11211** [comment]
@ltagliaferri. I was writing a java program to reproduce the error in the post and it doesn't seem to fit in the box, like it won't format properly, I have uploaded a ss -
![image](https://user-images.githubusercontent.com/91385411/139568121-e4071c38-5b3c-49b1-9190-8f539753bebe.png)

here is the HTML view-
![image](https://user-images.githubusercontent.com/91385411/139568218-aa57e3d9-1a3b-4eb0-86f8-7e9f104284a5.png)

P.S- _I apologize for naming my file "yeah", I'll rename it once I am done with editing it._

**Comment by ltagliaferri** [comment]
Try putting a space (as in, press the spacebar) in the blank line

**Comment by ltagliaferri** [comment]
If you keep having the issue feel free to open the pull request and we can work on it internally

**Comment by Ln11211** [comment]
@ltagliaferri , is my previous post merged and present in the main repo ?, because I have a conflict with my version of timeline, I mean the last time I fetched upstream was long ago so I am encountering a problem like this -
![image](https://user-images.githubusercontent.com/91385411/139714054-025daf80-ab4c-4c17-b6cf-3cf8a53152b1.png)

If I am not wrong, I would be opening a PR in my Repository(Ln11211/learn) right ?, so if I open the PR and Merge changes and commits(the commits that I am behind), would that fix my timeline ?

![image](https://user-images.githubusercontent.com/91385411/139713954-a7396082-6a6d-4be8-8b6d-7871fbc2fc7b.png)
**OR ELSE**
If my previous post is present in the main repository, then I will delete my current repository(that has conflicts and an old timeline) and re-fork a new one, but I am wondering what would happen to the PR that is still not merged if I delete my present repository ?

PS:- I cannot fetch upstream and merge commits, the Github is asking me to raise a PR to resolve the conflicts. I am done writing the second post too and I will raise a PR once this is resolved. I am sorry for the mess.

**Comment by ltagliaferri** [comment]
Can you try the following: 

1. Pull the tutorial out of the repo (put it in a temporary folder or your desktop for instance)
2. Make it so you will not commit any new changes to sourcegraph / learn (remove any other files that may be changing things)
3. Pull from the upstream so that your local repo matches sourcegraph / learn
4. Put the file(s) you moved back into the relevant directory
5. Open the PR on this project

**Comment by Ln11211** [comment]
well, I haven't yet placed my tutorial in the posts folder(I have it on my desktop), the problem seems to come from my other post #303, I have conflicts in that and only people with write access can resolve them, well your plan might work but I am 17 commits ahead and 9 commits behind, the 9 commits are from the past 3 days, If you can resolve the conflict in my previous PR, would that solve the problem?

How about I fork it to another account and then simply copy all the contents from there and paste them here in my present repo ?

**Comment by Ln11211** [comment]
> Can you try the following:
> 
> 1. Pull the tutorial out of the repo (put it in a temporary folder or your desktop for instance)
> 2. Make it so you will not commit any new changes to sourcegraph / learn (remove any other files that may be changing things)
 
Oh well, are you talking about my previous tutorial file ?, well then I can try it out with that one and it might work..

**Comment by Ln11211** [comment]
@ltagliaferri, sorry to ping you a lot, I tried moving the files(changes) and that was hard, I mean it's hard to determine which file should I move and then every time I move files it's asking me to commit to my repo. I got a plan that we can try that I think will hopefully save me, it goes like this-
1. We will resolve the conflicts in my previous PR and merge it so that the necessary file(my previous tutorial )is now in the main repository.
2. Then I will delete my repository and fork it again that way my repository is fresh and has no conflicts :)

_Consider this_
Let's say hypothetically I delete my Repository now, and fork a new one, would that mean I no longer have to pull and merge upstream right?(I will no longer have these conflicts). BUT I more worried about the previous PR, so let's say I delete the repo, so when a repo doesn't exist will the corresponding PR be erased too? Well If I make a new repo and then copy paste my files(this tutorial), wouldn't that solve the problem ?

Hey if we don't arrive at a solution(hopefully we will), I pasted the tutorial file in the pastepin [here](https://pastebin.com/1DkJ8626). It is protected with a password and only people with that link can open it, the password to open the file is "learn". I'm sorry that I caused a havoc, It's 'scary' and at the same time 'exciting' working with the git....
